controllers/bsl.go

```diff
@@ -418,14 +418,10 @@ func (r *DPAReconciler) ensureSecretDataExists(dpa *oadpv1alpha1.DataProtectionA
 		}
 		// Check if the BSL secret key configured in the DPA exists with a secret data
 		secretName, secretKey := r.getSecretNameAndKeyforBackupLocation(*bsl)
-		bslSecret, err := r.getProviderSecret(secretName)
+		err := r.verifySecretContent(secretName, secretKey)
```

Comment:
I like it, but we could have even further optimization of this code.

The following function cab be modified to do the validation, right?

```
getSecretNameAndKeyforBackupLocation
```

Reply:
yes, I can modify this function something like this:
`func (r *DPAReconciler) validateSecretforBackupLocation(bslspec oadpv1alpha1.BackupLocation) (error) {

	var secretName, secretKey string
	if bslspec.CloudStorage != nil {
		if bslspec.CloudStorage.Credential != nil {
			secretName = bslspec.CloudStorage.Credential.Name
			secretKey = bslspec.CloudStorage.Credential.Key
		}
	}
	if bslspec.Velero != nil {
		secretName, secretKey = r.getSecretNameAndKey(bslspec.Velero.Config, bslspec.Velero.Credential, oadpv1alpha1.DefaultPlugin(bslspec.Velero.Provider))
	}

	err := r.verifySecretContent(secretName, secretKey)
	if err != nil {
		return err
	}

	return nil
}`


but this function is being used at many places to get secret Name and Key, like in the ReconcileBackupStorageLocation function as well.
Modifying it further will create some more mess. 

@mpryc

Reply:
Getting and validating secret by it's name and key should be the same across all OADP places as it's always be the same result: `secretName` and `secretKey` existing and valid or not. So I don't think it's actually wrong approach to have unique common function to do that.

Reply:
@mpryc Can you please have a look? I have tried to further simplify it.